AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Core/Promise.swift

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+struct Promise<Output, Failure: Error> {
```

Comment:
Why not use [Future](https://developer.apple.com/documentation/combine/future)? If any additional functionality provided here merits this class, it may be good adding some swiftdoc explaining that?

Reply:
I was trying to use Future, but it only available since iOS 13. The supporting iOS version for v1 is iOS 11. I have to implemente it manually.

Reply:
will add doc to explain it 👍